tests/logictest/suites/mode/standalone/explain/where_optimizer.test

```diff
@@ -0,0 +1,46 @@
+statement ok
```

Comment:
Need more test cases, such as the following
```sql
mysql> create table if not exists t_where_optimizer (a int, b int);
ERROR 2013 (HY000): Lost connection to MySQL server during query
No connection. Trying to reconnect...
Connection id:    9
Current database: *** NONE ***

Query OK, 0 rows affected (0.04 sec)

mysql> explain select a from t_where_optimizer where sum(a) > 1;
+-----------------------------------------------------------------------------------------------------------+
| explain                                                                                                   |
+-----------------------------------------------------------------------------------------------------------+
| Project: [a (#0)]                                                                                         |
| └── Filter: [sum(a) > 1]                                                                                  |
|     └── Scan: default.default.t_where_optimizer, filters: [sum(a) > 1], Sort: [none], limit: [none]       |
+-----------------------------------------------------------------------------------------------------------+
3 rows in set (0.04 sec)
Read 0 rows, 0.00 B in 0.003 sec., 0 rows/sec., 0.00 B/sec.

mysql> select a from t_where_optimizer where sum(a) > 1;
ERROR 1105 (HY000): Code: 1015, displayText = Unsupported physical scalar: AggregateFunction(AggregateFunction { display_name: "sum(a)", func_name: "sum", distinct: false, params: [], args: [BoundColumnRef(BoundColumnRef { column: ColumnBinding { database_name: Some("default"), table_name: Some("t_where_optimizer"), column_name: "a", index: 0, data_type: Int32(Int32), visible_in_unqualified_wildcard: true } })], return_type: Int64(Int64) }).
mysql>
```

Reply:
another case
```
mysql> explain select a from t_where_optimizer where abs(a) > 1;
+----------------------------------------------------------------------------------------------------------------------------------+
| explain                                                                                                                          |
+----------------------------------------------------------------------------------------------------------------------------------+
| Project: [a (#0)]                                                                                                                |
| └── Filter: [abs(t_where_optimizer.a (#0)) > 1]                                                                                  |
|     └── Scan: default.default.t_where_optimizer, filters: [abs(t_where_optimizer.a (#0)) > 1], Sort: [none], limit: [none]       |
+----------------------------------------------------------------------------------------------------------------------------------+
3 rows in set (0.03 sec)
Read 0 rows, 0.00 B in 0.002 sec., 0 rows/sec., 0.00 B/sec.
```

Reply:
> select a from t_where_optimizer where sum(a) > 1;`

This sql is invalid. You mean `select a from t_where_optimizer group by a having sum(a) > 1;` ? This kind of sql will not be optimized by `WhereOptimizer`.

Reply:
I think almost every case is covered by other tests in stateless-test and sqllogictest.

Reply:
> another case
> explain select a from t_where_optimizer where abs(a) > 1;

In the current prewhere optimizaiton rule, we will not move the expression involved all columns to prewhere. In this case, we only need to scan column a, and where clause only involves column a, so we it will not be moved to prewhere.

Such case is already covered, like: `explain select a from t_where_optimizer where a = 1;`.

Reply:
> > select a from t_where_optimizer where sum(a) > 1;
> 
> This sql is invalid. You mean `select a from t_where_optimizer group by a having sum(a) > 1;` ? This kind of sql will not be optimized by `WhereOptimizer`.

Yes, it's invalid, so we don't need to consider aggregation function in `collect_columns_no_subqueries`?

Reply:
> In the current prewhere optimizaiton rule, we will not move the expression involved all columns to prewhere

Why we won't move the expression involved all columns to prewhere?

Reply:
Because it's meaningless. For such case, prewhere will degenerate to normal `Filter - Scan`. So we can just follow the flow without prewhere.

```
   Prewhere
     ┌──►Pre Read Data────►Filter───►Read Remaining Columns and Merge Chunks
     │                                               │
     │                                               │
Start│                                               │
     │                                               │
     │                                               ▼
     └──►Read All Columns─────────────────────►Deserialize ─────►Generate
 No Prewhere
```